### PR TITLE
payloadsallthethings: init at 3.0-unstable-2024-01-21

### PR DESCRIPTION
**Pull request by shard77**
## Description of changes
PayloadsAllTheThings - A list of useful payloads and bypass for Web Application Security and Pentest/CTF.

See:
- https://github.com/swisskyrepo/PayloadsAllTheThings
- https://repology.org/project/payloadsallthethings/information

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/pa/payloadsallthethings/package.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    repo = "PayloadsAllTheThings";
+    rev = "97cfeee270395a838802fa1fcb8a4d5ffc6d6b48";
+    hash = "sha256-LRS60v0o5nPSLfGFH6P0Y5roN8Mk5/KyRF4SWTv/7Hw=";
+  };
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/share/payloadsallthethings
+    cp -a * $out/share/payloadsallthethings
+    rm -f $out/share/payloadsallthethings/{CONTRIBUTING.md,mkdocs.yml,custom.css}
```
Comment (D3vil0p3r) [inline pkgs/by-name/pa/payloadsallthethings/package.nix:21]:
```suggestion
    rm CONTRIBUTING.md mkdocs.yml custom.css
    cp -a * $out/share/payloadsallthethings
```

**Review by D3vil0p3r — CHANGES_REQUESTED** [review]
Just invert the lines to avoid the copy of useless files.

**Review by D3vil0p3r — APPROVED** [review]
Now LGTM

File: pkgs/by-name/pa/payloadsallthethings/package.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{ lib
+, fetchFromGitHub
+, stdenvNoCC
+}:
+
+stdenvNoCC.mkDerivation {
+  pname = "payloadsallthethings";
+  version = "1894.97cfeee";
```
Comment (D3vil0p3r) [inline pkgs/by-name/pa/payloadsallthethings/package.nix:8]:
```suggestion
  version = "unstable-2024-01-21";
```

**Review by D3vil0p3r — CHANGES_REQUESTED** [review]
Change also title and commit message accordingly

**Review by D3vil0p3r — APPROVED** [review]

File: pkgs/by-name/pa/payloadsallthethings/package.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/share/payloadsallthethings
+    rm CONTRIBUTING.md mkdocs.yml custom.css
+    cp -a * $out/share/payloadsallthethings
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "A list of useful payloads and bypass for Web Application Security and Pentest/CTF";
```
Comment (AndersonTorres) [inline pkgs/by-name/pa/payloadsallthethings/package.nix:26]: where is homepage?

Comment (shard77) [reply pkgs/by-name/pa/payloadsallthethings/package.nix:27]: added it

**Review by AndersonTorres — APPROVED** [review]

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 283621` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>payloadsallthethings</li>
  </ul>
</details>

**Pull request merged**